README.md

```diff
@@ -1,8 +1,110 @@
+# Overview
+
+The samples operator manages the sample imagestreams and templates stored in the openshift namespace, and any docker credentials, stored as a secret, needed for the imagestreams to import the images they reference.
+
+On initial startup, the operator will create the default samples resource to initiate the creation of the imagestreams and templates.  Currently, the imagestreams are the CentOS based OKD imagestreams pointing to images on docker.io.  Similarly the templates are those categorized as OKD templates.  When details for obtaining the credentials for registry.redhat.io are finalized, we will begin defaulting to the RHEL and OCP imagestreams and templates.
+
+The samples operator, along with it's configuration resources, are contained within the openshift-cluster-samples-operator namespace. If an admin creates a secret with the name "openshift-samples" which contains the content of a docker config.json file in the operators namespace, it will be copied into the openshift namespace for use during imagestream import.
+
+The image for the samples operator contains imagestream and template definitions for the associated OpenShift release. Each sample includes an annotation that denotes the OpenShift version that it is compatible with. The operator uses this annotation to ensure that each sample matches it's release version. Samples outside of its inventory are ignored, as are skipped samples (see below). Modifications to any samples that are managed by the operator will be reverted automatically.
+
+
+
+The samples resource includes a finalizer which will clean up the following upon it's deletion:
+
+- Operator managed imagestreams
+- Operator managed templates
+- Operator generated configuration resources
+- Cluster status resources
+- An openshift-samples secret (if supplied)
+
+
+
+Upon deletion of the samples resource, the samples operator will recreate the resource using the default configuration.
+
+# Configuration
+
+The samples resource offers the following configuration fields:
+
+- ManagementState
+-- Managed: the operator will update the samples as the configuration dictates
+-- Unmanaged: the operator will ignore updates to the samples resource object and any imagestreams or templates in the openshift namespace
+-- Removed: the operator will remove the set of managed imagestreams and templates in the openshift namespace. It will ignore new samples created by the cluster admin or any samples in the skipped lists.  After the removals are complete, the operator will work like it is in the 'Unmanaged' state and ignore any watch events on the sample resources, secrets, imagestreams, or templates.  There are some caveats around concurrent creates and removal (see Change behaviors section).
+- Samples Registry
+-- Override the registry that images are imported from
+- Install type
+-- Use 'centos' for CentOS imagestreams and 'rhel' for RHEL imagestreams (and associatively between OCP and OKD content for the templates)
```

Comment:
s/OCP and OKD/OKD and OCP/  (keep the ordering consistent....  centos/rhel and okd/ocp)

Reply:
part of next push